src/AcceptanceTests/When_session_is_accessed_in_hosted_service_start.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace AcceptanceTests
+{
+    using Microsoft.Extensions.Hosting;
+    using NUnit.Framework;
+    using NServiceBus;
+    using Microsoft.Extensions.DependencyInjection;
+    using System.Threading.Tasks;
+    using System.Threading;
+    using System;
+
+    [TestFixture]
+    public class When_session_is_accessed_in_hosted_service_start
+    {
+        [Test]
+        public async Task Should_be_available_when_configured_after_NServiceBus()
+        {
+            var host = Host.CreateDefaultBuilder()
+                .UseNServiceBus(hostBuilderContext =>
+                {
+                    var endpointConfiguration = new EndpointConfiguration("MyEndpoint");
+                    endpointConfiguration.SendOnly();
+                    endpointConfiguration.UseTransport<LearningTransport>();
+                    return endpointConfiguration;
+                })
+                .ConfigureServices((ctx, serviceProvider) => serviceProvider.AddHostedService<HostedServiceThatAccessSessionInStart>())
+                .Build();
+
+            await host.StartAsync();
+            await host.StopAsync();
+        }
+
+        [Test]
+        public void Should_throw_when_hosted_service_is_configured_before_NServiceBus()
+        {
+            var ex = Assert.ThrowsAsync<InvalidOperationException>(async () =>
+            {
+                var host = Host.CreateDefaultBuilder()
+                .ConfigureServices((ctx, serviceProvider) => serviceProvider.AddHostedService<HostedServiceThatAccessSessionInStart>())
+                .UseNServiceBus(hostBuilderContext =>
+                  {
+                      var endpointConfiguration = new EndpointConfiguration("MyEndpoint");
+                      endpointConfiguration.SendOnly();
+                      endpointConfiguration.UseTransport<LearningTransport>();
+                      return endpointConfiguration;
+                  })
+                .Build();
+
+                await host.StartAsync();
```

Comment:
This test doesn't stop the host, should that be added?

Reply:
since this is expected to throw the stop would never be called?

Reply:
Agreed here stopp is not needed